PackageExplorer/Utilities/NuGetExtensions.cs

```diff
@@ -0,0 +1,271 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace NuGet.Frameworks
+{
+	/// <summary>
+	/// Class for Extensions related to NuGetFramework objects.
+	/// </summary>
+	public static class NuGetFrameworkExtensions
```

Comment:
I'll add more comments -- these were refactored into extensions specifically because the current NuGet framework is very convoluted in how it organizes its internal data structures.